src/AppInstallerCLICore/Workflows/SourceFlow.cpp

```diff
@@ -82,19 +84,31 @@ namespace AppInstaller::CLI::Workflow
 
     void AddSource(Execution::Context& context)
     {
-        std::string name(context.Args.GetArg(Args::Type::SourceName));
-        std::string arg(context.Args.GetArg(Args::Type::SourceArg));
-        std::string type;
+        Repository::SourceDetails sourceDetails;
+        sourceDetails.Name = context.Args.GetArg(Args::Type::SourceName);
+        sourceDetails.Arg = context.Args.GetArg(Args::Type::SourceArg);
         if (context.Args.Contains(Args::Type::SourceType))
         {
-            type = context.Args.GetArg(Args::Type::SourceType);
+            sourceDetails.Type = context.Args.GetArg(Args::Type::SourceType);

```

Comment:
>sourceDetails.Type [](http://example.com/codeflow?start=12&length=18)

The default type is the preindexed packaged source, since we are construcing the SourceDetails here, shall we assign the default here?